packages/components/rating/src/rating-segment.tsx

```diff
@@ -0,0 +1,119 @@
+import {useMemo} from "react";
+import {clsx, dataAttr} from "@nextui-org/shared-utils";
+import {useHover} from "@react-aria/interactions";
+import {Radio} from "@nextui-org/radio";
+
+import {useRatingContext} from "./rating-context";
+import {RatingIcon} from "./rating-icon";
+
+interface RatingSegmentProps {
+  index: number;
+  icon?: React.ReactNode;
+  fillColor?: string;
+}
+
+const RatingSegment = ({index, icon, fillColor}: RatingSegmentProps) => {
+  const context = useRatingContext();
+  const {
+    ratingValue,
+    isRTL,
+    isIconWrapperHovered,
+    shouldConsiderHover,
+    precision,
+    slots,
+    classNames,
+    isSingleSelection,
+    name,
+    onChange,
+    onBlur,
+    setRatingValue,
+  } = context;
+
+  let value = ratingValue.selectedValue;
+
+  if (isIconWrapperHovered && shouldConsiderHover) {
+    value = ratingValue.hoveredValue;
+  }
+
+  const calculateOffsets = (value: number, index: number, isSingleSelection: boolean) => {
+    if (isSingleSelection) {
+      const singleOffset = Number(Math.floor(value) - 1 == index);
+
+      return {offset: singleOffset, offsetRTL: 1 - singleOffset};
+    }
+
+    if (Math.floor(value) > index) {
+      return {offset: 1, offsetRTL: 0};
+    }
+
+    if (Math.floor(value) < index) {
+      return {offset: 0, offsetRTL: 1};
+    }
+
+    const fractionalPart = value - Math.floor(value);
+
+    return {offset: fractionalPart, offsetRTL: 1 - fractionalPart};
+  };
+
+  const {offset, offsetRTL} = calculateOffsets(value, index, isSingleSelection);
+
+  const segmentStyles = slots.iconSegment({class: clsx(classNames?.iconSegment)});
+  const {isHovered, hoverProps} = useHover({});
+
+  const radioButtons = useMemo(() => {
+    const numButtons = Math.floor(1 / precision);
+    const gridStyle = {
+      display: "grid",
+      gridTemplateColumns: `repeat(${numButtons}, 1fr)`,
+    };
+
+    return (
+      <div
+        className={slots.radioButtonsWrapper({class: classNames?.radioButtonsWrapper})}
+        style={gridStyle}
+      >
+        {Array.from(Array(numButtons)).map((_, idx) => {
+          const radioButtonValue =
+            idx === numButtons - 1 ? index + 1 : index + precision + idx * precision;
+
+          return (
+            <div
+              key={idx}
+              className={slots.radioButtonWrapper({class: classNames?.radioButtonWrapper})}
+              onMouseMove={() => {
+                setRatingValue({
+                  hoveredValue: radioButtonValue,
+                  selectedValue: ratingValue.selectedValue,
+                });
+              }}
+            >
+              <Radio
+                key={idx}
+                classNames={{base: "w-full h-full m-0"}}
+                data-slot="radio"
+                name={name}
+                value={radioButtonValue.toString()}
+                onBlur={onBlur}
+                onChange={onChange}
+              />
```

Comment:
Each radio lacks an `accName`, making it inaccessible for assistive technology users. 
Also, please ensure the Rating component accepts an overall label. 
This document might be helpful. 
https://mui.com/material-ui/react-rating/#accessibility

Reply:
Done in: [374136d](https://github.com/nextui-org/nextui/pull/3808/commits/374136dec9ee9ecc498ad02463d1cd0cc0fa66c7)